src/main/java/net/minecraftforge/network/PlayMessages.java

```diff
@@ -109,7 +111,17 @@ public static void encode(SpawnEntity msg, FriendlyByteBuf buf)
             buf.writeShort(msg.velZ);
             if (msg.entity instanceof IEntityAdditionalSpawnData)
             {
-                ((IEntityAdditionalSpawnData) msg.entity).writeSpawnData(buf);
+                final FriendlyByteBuf spawnDataBuffer = new FriendlyByteBuf(Unpooled.buffer());
+
+                ((IEntityAdditionalSpawnData) msg.entity).writeSpawnData(spawnDataBuffer);
+
+                buf.writeVarInt(spawnDataBuffer.readableBytes());
+                buf.writeBytes(spawnDataBuffer);
+
+                spawnDataBuffer.release();
+            }
+            else {
```

Comment:
Formatting: braces on newlines.
```suggestion
            else
            {
```